File: packages/zodiac-dart/lib/src/models/ruling_planet.dart
/// Ruling planet model.
///
/// Represents a celestial body used as a ruling planet in Western astrology.
class RulingPlanet {
  /// Stable identifier (e.g., "mars").
  final String id;

  /// English display name.
  final String name;

  /// Unicode symbol (if applicable).
  final String? symbol;

  const RulingPlanet({
    required this.id,
    required this.name,
    this.symbol,
  });
}

File: packages/zodiac-js/src/models/ruling-planet.ts
/**
 * Ruling planet model.
 *
 * Represents a celestial body used as a ruling planet in Western astrology.
 */
export interface RulingPlanet {
  /** Stable identifier (e.g., "mars") */
  readonly id: string;

  /** English display name */
  readonly name: string;

  /** Unicode symbol (if applicable) */
  readonly symbol?: string;
}
